caddyconfig/httpcaddyfile/directives.go

```diff
@@ -119,6 +119,7 @@ type Helper struct {
 	matcherDefs  map[string]caddy.ModuleMap
 	parentBlock  caddyfile.ServerBlock
 	groupCounter counter
+	tlsCertTags  map[string]string
```

Comment:
Ah, this could work, but only for this one thing. A more general and useful approach would be to have a field defined like `state map[string]interface{}` that we could assign `tlsCertTags` into, for example: `h.state["tls_cert_tags"] = tlsCertTags` and then get it out with a type assertion.

This way, other directives can use the state to remember things while they set up the JSON.

Reply:
Moved to https://github.com/caddyserver/caddy/pull/3111